### PR TITLE
Add an endpoint to return TSD Token

### PR DESCRIPTION
**Pull request by a-ghanem**
### Describe the pull request:

- [x] Functional change

### Pull request long description:
Added a new REST API endpoint to return a TSD token.

### CODE REVIEW

**Comment by kjellp** [comment]
@joshbaskaran do you think this PR is possible to get updated, reviewed and merged relatively quickly, or do you foreseen some trouble here? It would possibly speed up uploads quite a bit, and maybe get us beyound the 2hrs token expiry problem we're facing currently.

**Comment by joshbaskaran** [comment]
@kjellp 
I would prefer to change the end-point to `/tsd/token` instead.
Other than that I'm quite sure how this would resolve a token expiry issue. I would like some context here like who/what is consuming this end-point and it used.

**Comment by kjellp** [comment]
The name change of end-point should not be a show-stopper, only need to update our lega-commander code to use that instead (currently the -b option of upload command is using the gettoken endpoint of the proxy service.

After lc swaps the LS AAI token for a TSD token, that TSD token has a TTL of ~24 hrs instead of ~2 hrs for the LS AAI token. But we need to double check that this also holds for a folder upload with many files (and that not a LS AAI token based "init" communication with the proxy service is needed for each of the files in the folder upload). 

And to enable this "direct TSD upload pattern" using the -b option and the gettoken endpoint, also requires a closer inspection of the resume functionality, that we don't immediately break that.... (it's working for individual files now (via proxy service uploads), but not for folders...)

File: src/main/java/no/uio/ifi/ltp/controllers/rest/ProxyController.java [hunk 1]
```diff
@@ -158,4 +158,16 @@ protected String getElixirAAIToken(String bearerAuthorization) {
         return bearerAuthorization.replace("Bearer ", "");
     }
 
+    /**
+     * Gets TSD token.
+     *
+     * @param bearerAuthorization Elixir AAI token.
+     * @return TSD token.
+     */
+    @GetMapping("/gettoken")
```
Comment (joshbaskaran) [inline src/main/java/no/uio/ifi/ltp/controllers/rest/ProxyController.java:167]:
```suggestion
    @GetMapping("/tsd/token")
```

**Review by joshbaskaran — DISMISSED** [review]

**Comment by joshbaskaran** [comment]
@kjellp 
In that case, please feel free to merge the branch :)

File: src/main/java/no/uio/ifi/ltp/controllers/rest/ProxyController.java (hunk 1 above)
Comment (kjellp) [reply src/main/java/no/uio/ifi/ltp/controllers/rest/ProxyController.java:167]: We (Parisa and me) devcided it's best to do the change of the end-point and the lega-commander client in sync. So we approve this PR without doing the end-point name change, and create a new card in our Project board to solve these to required changed (lc and proxy) in sync.

**Comment by Parisa68** [comment]
@dependabot rebase

**Pull request merged**